core/middlewares/with-routes.ts

```diff
@@ -205,6 +205,10 @@ const updateStatusCache = async (
 };
 
 const clearLocaleFromPath = (path: string, locale: string) => {
+  if (path === `/${locale}`) {
+    return '/';
+  }
+
   if (path.startsWith(`/${locale}/`)) {
```

Comment:
The existing condition wasn't catching localized home page routes that didn't end with a trailing slash, e.g. `/fr`.